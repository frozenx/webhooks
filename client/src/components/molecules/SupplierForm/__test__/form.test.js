import React from 'react';
import { shallow } from 'enzyme';
import DetailForm from '../Form';

describe('form component test cases', () => {
  let Wrapper;
  let event = {
    preventDefault: jest.fn(),
    stopPropagation: jest.fn()
  }
  beforeEach(() => {
    Wrapper = shallow(<DetailForm />);
  });

  test('render the component', () => {
    expect(Wrapper).toMatchSnapshot();
  });
  
  test('click on save and cancel should shot the defalut behaviour', () => {

    Wrapper.find('.submit-btn').simulate('click', event);
    Wrapper.find('#save').simulate('click', event);
    expect(event.preventDefault).toHaveBeenCalledTimes(2);
    expect(event.stopPropagation).toHaveBeenCalledTimes(2);
  });
});
