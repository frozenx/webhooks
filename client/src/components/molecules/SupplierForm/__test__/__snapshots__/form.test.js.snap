// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form component test cases render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SupplierDetailForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <GenericFormContainer />,
        <ForwardRef>
          <_default
            className="submit-btn"
            id="save-as-draft"
            onClick={[Function]}
            type="submit"
            variant="secondary"
          >
            Cancel
          </_default>
          <_default
            className="submit - btn enabled"
            id="save"
            onClick={[Function]}
            type="submit"
            variant="primary"
          >
             Save
          </_default>
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "genericFormData": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <_default
              className="submit-btn"
              id="save-as-draft"
              onClick={[Function]}
              type="submit"
              variant="secondary"
            >
              Cancel
            </_default>,
            <_default
              className="submit - btn enabled"
              id="save"
              onClick={[Function]}
              type="submit"
              variant="primary"
            >
               Save
            </_default>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Cancel",
              "className": "submit-btn",
              "id": "save-as-draft",
              "onClick": [Function],
              "type": "submit",
              "variant": "secondary",
            },
            "ref": null,
            "rendered": "Cancel",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": " Save",
              "className": "submit - btn enabled",
              "id": "save",
              "onClick": [Function],
              "type": "submit",
              "variant": "primary",
            },
            "ref": null,
            "rendered": " Save",
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htoDjs",
            "isStatic": true,
            "rules": Array [
              "
    button{
        margin-right: 20px;
    }

",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htoDjs",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-jzJRlG",
        "isStatic": true,
        "rules": Array [
          "
    margin-bottom : 24px;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-jzJRlG",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <GenericFormContainer />,
          <ForwardRef>
            <_default
              className="submit-btn"
              id="save-as-draft"
              onClick={[Function]}
              type="submit"
              variant="secondary"
            >
              Cancel
            </_default>
            <_default
              className="submit - btn enabled"
              id="save"
              onClick={[Function]}
              type="submit"
              variant="primary"
            >
               Save
            </_default>
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "genericFormData": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <_default
                className="submit-btn"
                id="save-as-draft"
                onClick={[Function]}
                type="submit"
                variant="secondary"
              >
                Cancel
              </_default>,
              <_default
                className="submit - btn enabled"
                id="save"
                onClick={[Function]}
                type="submit"
                variant="primary"
              >
                 Save
              </_default>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Cancel",
                "className": "submit-btn",
                "id": "save-as-draft",
                "onClick": [Function],
                "type": "submit",
                "variant": "secondary",
              },
              "ref": null,
              "rendered": "Cancel",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": " Save",
                "className": "submit - btn enabled",
                "id": "save",
                "onClick": [Function],
                "type": "submit",
                "variant": "primary",
              },
              "ref": null,
              "rendered": " Save",
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htoDjs",
              "isStatic": true,
              "rules": Array [
                "
    button{
        margin-right: 20px;
    }

",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htoDjs",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-jzJRlG",
          "isStatic": true,
          "rules": Array [
            "
    margin-bottom : 24px;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-jzJRlG",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
